src/bgg/circuit/utils.rs

```diff
@@ -1,40 +1,50 @@
-use crate::bgg::circuit::Evaluable;
-
 use super::PolyCircuit;
+use crate::bgg::circuit::Evaluable;
 
-/// Build a circuit that takes as input a public circuit and a private input, and outputs inner
-/// products between the `num_priv_input`-sized output of the public circuit and the private input.
-pub fn build_circuit_ip_priv_and_pub_outputs<E: Evaluable>(
```

Comment:
@enricobottazzi Can you keep the original function name as it should be more generic than fhe decryption?
Also, I don't think this function needs to restrict the the concrete private input size.

Reply:
What would be a case in which this:
1. is used for a more generic scope than fhe decryption
2. the private input size is larger than 1

Reply:
Our current implementation does not have concrete codes for both cases, but we (or any other developers using our implementation) might use a public circuit which handles some inner product for the private input larger than 1 in the future.
I don't think we need to restrict the generality of this function, but do you think there is some drawback due to this generality?